website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -9,11 +9,186 @@ description: >-
+  - `Gateway.read`
+  - `ReferencePolicy.create`
+- (Optional) The [jq](https://stedolan.github.io/jq/download/) command line processor for JSON can be installed, which will ease route retrieval during the upgrade process.
+
+### Procedure
+
+
+1. Verify the current version of the `consul-api-gateway-controller` `Deployment`:
+
+  ```shell-session
+  $ kubectl get deployment --namespace consul consul-api-gateway-controller --output=jsonpath= "{@.spec.template.spec.containers[?(@.name=='api-gateway-controller')].image}"
+  ```
+
+  You should receive the following response:
+
+  ```log
+  "hashicorp/consul-api-gateway:0.2.1"
+  ```
+
+1. Retrieve all gateways that have a secret in a different namespace. If you have installed the [`jq`](https://stedolan.github.io/jq/) utility, you can skip to [step 4](#jq-command-secrets). Otherwise, issue the following command to get all `Gateways` across all namespaces:
+
+  ```shell-session
+  $ kubectl get Gateway --output json --all-namespaces
+  ```
+
+
+  If you have any active `Gateways`, you will receive output similar to the following response. The output has been truncated to show only relevant fields:
+
+  ```yaml
+    apiVersion: v1
+    items:
+    - apiVersion: gateway.networking.k8s.io/v1alpha2
+      kind: HTTPRoute
+      metadata:
+        name: example-http-route,
+        namespace: example-namespace,
+        ...
+      spec:
+        parentRefs:
+        - group: gateway.networking.k8s.io
+          kind: Gateway
+          name: gateway
+          namespace: gw-ns
+        rules:
+        - backendRefs:
+          - group: ""
+            kind: Service
+            name: web-backend
+            namespace: gateway-namespace
+            ...
+          ...
+    - apiVersion: gateway.networking.k8s.io/v1alpha2
+      kind: TCPRoute
+      metadata:
+        name: example-tcp-route,
+        namespace: a-different-namespace,
+        ...
+      spec:
+        parentRefs:
+        - group: gateway.networking.k8s.io
+          kind: Gateway
+          name: gateway
+          namespace: gateway-namespace
+        rules:
+        - backendRefs:
+          - group: ""
+            kind: Service
+            name: web-backend
+            namespace: gateway-namespace
+            ...
+      ...
+  ```
+
+1. Inspect the `secret` entries for each of the routes.
+
+  If a `namespace` field is not defined in the `secret` or if the namespace matches the namespace of the parent `Gateway`, then no additional action is required for the `secret`. Otherwise, note the `namespace` field values for `secret` configurations that have a `namespace` defined that do not match the namespace of the parent `Gateway`. You must also note the `namespace` of the parent gateway. You will need these to create a `ReferencePolicy` that explicitly allows each cross-namespace secret-to-gateway pair to prevent the route from breaking (see [step 5](#create-secret-reference-grant)).
```

Comment:
```suggestion
  If a `namespace` field is not defined in the `secret` or if the namespace matches the namespace of the parent `Gateway`, then no additional action is required for the `secret`. Otherwise, note the `namespace` field values for `secret` configurations that have a `namespace` defined that do not match the namespace of the parent `Gateway`. You must also note the `namespace` of the parent gateway. You will need these to create a `ReferencePolicy` that explicitly allows each cross-namespace secret-to-gateway pair to prevent the route from breaking (see [step 5](#create-secret-reference-policy)).
```